contracts/Whitelist.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.6.10;
+
+import "./packages/oz/Ownable.sol";
+
+/**
+ * @author Opyn Team
+ * @title Whitelist Module
+ * @notice The whitelist module keeps track of all valid Otoken contracts.
+ */
+contract Whitelist is Ownable {
+    mapping(bytes32 => bool) internal whitelistedProduct;
+
+    event ProductWhitelisted(
+        bytes32 productHash,
+        address indexed underlying,
+        address indexed strike,
+        address indexed collateral
+    );
+
+    /**
+     * @notice check if a product is supported
+     * @dev product = the hash of underlying, collateral and strike asset
+     * @param _underlying option underlying asset address
+     * @param _strike option strike asset address
+     * @param _collateral option collateral asset address
+     * @return boolean, true if product is supported
+     */
+    function isWhitelistedProduct(
+        address _underlying,
+        address _strike,
+        address _collateral
+    ) external view returns (bool) {
+        bytes32 productHash = keccak256(abi.encode(_underlying, _strike, _collateral));
+
+        return whitelistedProduct[productHash];
+    }
+
+    /**
+     * @notice whitelist a product
```

Comment:
@notice: Allows the owner of the whitelistModule to whitelist a new product.

Reply:
I think the @notice should just give a clear simple def of function, and @dev any more technical/implementation details. I made it clear in the @dev that this can only be called from owner. @antoncoding thoughts ?

Reply:
yeah we should keep the @notice simple, but I think Aparna's suggestion is valid and also not too technical so we can use that. 
btw we don't have to call it whitelist "module", just whitelist.

Reply:
I imagine market makers / people who aren't coders will be reading @notice. Making that as clear as possible is really important from that perspective! Also whatever we have in @notice can be copy pasted into our documentation.

Reply:
cool will update that.